ipykernel/inprocess/client.py

```diff
@@ -114,7 +126,9 @@ async def execute(
         )
         msg = self.session.msg("execute_request", content)
         await self._dispatch_to_kernel(msg)
-        return msg["header"]["msg_id"]
+        res = msg["header"]["msg_id"]
+        assert isinstance(res, str)
```

Comment:
I wondered if `typing.cast` could be safer, but per protocol it looks like a valid assertion

Reply:
I agree, my hope is that if it's wrong we can at least trigger failures with nightlies/pre-release, whcih a cat would not